src/lib/rootfs/rootfs.c

```diff
@@ -245,18 +254,22 @@ int singularity_rootfs_check(void) {
 int singularity_rootfs_chroot(void) {
     
     singularity_priv_escalate();
-    singularity_message(VERBOSE, "Entering container file system root: %s\n", joinpath(mount_point, OVERLAY_FINAL));
-    if ( chroot(joinpath(mount_point, OVERLAY_FINAL)) < 0 ) { // Flawfinder: ignore (yep, yep, yep... we know!)
-        singularity_message(ERROR, "failed enter container at: %s\n", joinpath(mount_point, OVERLAY_FINAL));
+
+    if ( fchdir(singularity_rootfs_fd()) < 0 ) {
```

Comment:
Well, we aren't using chroot() to be secure as we will end up being the same user inside as outside, but we do need to pass lint'ing which I know generally checks for the chdir() soon after the chroot().

I don't see this step as actually gaining us anything.